pages/example-4-missing-authorization-gssp/vulnerable.tsx

```diff
@@ -13,7 +15,28 @@ export default function MissingAuthentication(props: Props) {
   return <Common {...props} />;
 }
 
-export async function getServerSideProps() {
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+
+  const sessionToken = context.req.cookies.sessionToken;
+
+  if (!sessionToken) {
+    return {
+      props: {
+        error: 'Session token not provided',
+      },
+    };
+  }
+
+  const user = await getUserByValidSessionToken(sessionToken);
+
+  if (!user) {
+    return {
+      props: {
+        error: 'Session token not valid',
+      },
+    };
+  }
+
```

Comment:
In this "vulnerable" example, I'm thinking that we want to have a small amount of code here (without any validation) because we see this sometimes in student projects.

Having a really short `getServerSideProps` and then asking the question "What is wrong here?" should be clearer to the beginners that we teach.

However, there is probably value in also asking "What is wrong here?" with your proposed example. 🤔

Maybe what we can do is to have 2 vulnerable examples - eg:

1. Rename this file to `vulnerable-1.tsx` (with the short, unmodified `getServerSideProps`)
2. Create a new file called `vulnerable-2.tsx`, with your changes

What do you think?

Reply:
agree =)

Reply:
oh, just had an idea how to combine your original weak code from vulnerable.tsx with the insufficien client-side check. will post a commit suggestion.

Reply:
closing this comment-threat because there is already a newer comment below